### PR TITLE
add validation for machine deployment scaling

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/15888.

### CODE REVIEW

File: pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go [hunk 1]
```diff
@@ -212,6 +217,25 @@ func (v *Validator) ValidateCluster(awsMachineDeployment infrastructurev1alpha2.
...
+
+	if min == 0 {
+		return microerror.Maskf(notAllowedError, "AWSMachineDeployment.Spec.Scaling.Min must not be 0.")
+	}
+
+	if max == 0 {
+		return microerror.Maskf(notAllowedError, "AWSMachineDeployment.Spec.Scaling.Max must not be 0.")
+	}
+
+	if min > max {
```
Comment (njuettner) [inline pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go:232]: I think it's the other way around max cannot be lower than min

**Review by njuettner — APPROVED** [review]
brain dead 🙈 ignore my one comment, all good 🚀

**Comment by anvddriesch** [comment]
I see this is already merged. However, don't we allow scaling to 0?
Or is that something that should only be done on update?

**Comment by xh3b4sd** [comment]
Good call. Maybe @paurosello knows use cases and we can adapt if this is something we want to do once in a while.

**Pull request merged**